<a href="https://colab.research.google.com/github/lovita11/DataScience/blob/main/Pertemuan2_LovitaPrasetia_250401020061.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
# Q1 Berapa jumlah baris dan kolom pada dataset Titanic? Kolom apa saja yang tersedia?
import pandas as pd
import numpy as np

# Muat Dataset Titanic
URL = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df = pd.read_csv(URL)

print(df.shape, df.columns.tolist())


(891, 12) ['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']


Pada dataset Titanic terdapat 891 baris dan 12 kolom.

Kolom-kolom yang ada pada dataset Titanic adalah :

1. PassengerId
2. Survived
3. Pclass
4. Name
5. Sex
6. Age
7. SibSp
8. Parch
9. Ticket
10. Fare
11. Cabin
12. Embarked

In [ ]:
# Q2 Kolom mana saja yang memiliki missing values? Berapa jumlah dan proporsinya (%) per kolom?

print(df.isnull().sum())

print(df.isnull().sum()/df.shape[0]*100)


PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64
PassengerId     0.000000
Survived        0.000000
Pclass          0.000000
Name            0.000000
Sex             0.000000
Age            19.865320
SibSp           0.000000
Parch           0.000000
Ticket          0.000000
Fare            0.000000
Cabin          77.104377
Embarked        0.224467
dtype: float64


Kolom yang memiliki missing value:
1. Age =  19.865320 %
2. Cabin = 77.104377 %
3. Embarked = 0.224467 %


In [ ]:
# Q3 Berapa persen penumpang yang selamat (Survived=1)?
df['Survived'].head()
jumlah_selamat = df['Survived'].sum()
total_penumpang = len(df)
presentase_selamat = (jumlah_selamat / total_penumpang) * 100
print(jumlah_selamat)
print(presentase_selamat)


342
38.38383838383838


Jumlah penumpang selamat : 342 orang

Persentase penumpang selamat : 38.38383838383838%

In [ ]:
# Q4 Berapa jumlahnya dan berapa rata- rata usianya?

wanita_kelas_1 = df[
    (df['Sex'] == 'female') &
     (df['Pclass'] == 1) &
    (df['Survived'] == 1)]
({"jumlah penumpang": [len (wanita_kelas_1)],
  "rata-rata usia": [wanita_kelas_1['Age'].mean()]
  } )


{'jumlah penumpang': [91], 'rata-rata usia': [np.float64(34.9390243902439)]}

Rata-rata usia pemumpang wanita kelas 1 : 91 orang

Rata-rata usia wanita kelas 1 : 34.9390243902439 Tahun

In [ ]:
# Q5 Gunakan groupby untuk menghitung tingkat keselamatan (mean Survived) per kelas penumpang. Kelas mana yang paling tinggi?

survival_rate = df.groupby('Pclass')['Survived'].mean() * 100
kelas_tertinggi = survival_rate.idxmax()
tingkat_keselamatan_tertinggi = survival_rate.max()
print(survival_rate)


Pclass
1    62.962963
2    47.282609
3    24.236253
Name: Survived, dtype: float64


Kelas penumpang yan memiliki keselamatan tertinggi = kelas 1

In [ ]:
# Q6 Buat analisis tambahan berdasarkan variabel lain yang Anda anggap menarik. Sertakan interpretasi singkat.
hasil = df.groupby('Sex').agg(
  total=('PassengerId','count'),
  selamat=('Survived','mean'))

print(hasil)



        total   selamat
Sex                    
female    314  0.742038
male      577  0.188908


Untuk penumpang yang paling banyak selamat adalah penumpang wanita karena penumpang wanita lebih diprioritaskan untuk evakuasi dari pada penumpang laki laki